java-strings/src/main/java/com/baeldung/string/WordIndexer.java

```diff
@@ -0,0 +1,46 @@
+package com.baeldung.string.searching;
+
+import java.util.ArrayList;
+
+public class WordIndexer {
+
+    public ArrayList findWord(String textString, String word) {
+        int index = -1;
+        ArrayList<Integer> indexes = new ArrayList<Integer>();
+        StringBuilder output = new StringBuilder();
+        boolean done = false;
+        String lowerCaseTextString = textString.toLowerCase();
+        String lowerCaseWord = word.toLowerCase();
+        int wordLength = word.length();
+
+        while(!done){
+            index = lowerCaseTextString.indexOf(lowerCaseWord, index + wordLength);  // Slight improvement
+            if (index != -1) {
+                indexes.add(index);
+            }else{
+                done = true;
+            }
+        }
+        return indexes;
+    }
+
+    public ArrayList findWordUpgrade(String textString, String word) {
+        int index = -1;
+        ArrayList<Integer> indexes = new ArrayList<Integer>();
+        StringBuilder output = new StringBuilder();
+        boolean done = false;
+        String lowerCaseTextString = textString.toLowerCase();
+        String lowerCaseWord = word.toLowerCase();
+        int wordLength = word.length();
+
+        while(!done){
+            index = lowerCaseTextString.indexOf(lowerCaseWord, index + wordLength);  // Slight improvement
```

Comment:
What happens if the string ends with the word you're searching for? Would the next iteration of the loop cause an error if index + wordLength is greater than the last position of the string? Be sure to test this scenario.

Reply:
Thank you, Kevin. They are all very good points. 
Java's indexOf() method gracefully responds to "... if the string ends with the word you're searching for" (i.e., no error should happen).  I added a JUnit test anyway to test that scenario.